### PR TITLE
OPTIONS Request does not return all allowed methods

### PR DESCRIPTION
**Pull request by webimpress**
**WIP** This is just basic test to demonstrate that it doesn't work correctly for all our route adapters.

The problem seems to be quite complex.

See discourse topic about it:
https://discourse.zendframework.com/t/strange-behaviour-of-implicitoptionsmiddleware-with-multiple-route-on-same-path/496

The problem also occurs in v2 😢

### CODE REVIEW

**Comment by weierophinney** [comment]
If it occurs in v2, we should likely fix it there as well. Can you rebase against master and update the branch against which this applies, please?

**Comment by weierophinney** [comment]
> Can you rebase against master and update the branch against which this applies, please?

Never mind; I forgot these middleware are not in master.

File: test/Middleware/ImplicitHeadMiddlewareTest.php [hunk 1]
```diff
@@ -127,9 +129,9 @@ public function testReturnsComposedResponseWhenPresentWhenRouteImplicitlySupport
     {
         $route = $this->prophesize(Route::class);
         $route->implicitHead()->willReturn(true);
-        $route->allowsMethod(RequestMethod::METHOD_GET)->willReturn(false);
 
         $result = $this->prophesize(RouteResult::class);
+        $result->getAllowedMethods()->willReturn([RequestMethod::METHOD_HEAD]);
```
Comment (weierophinney) [inline test/Middleware/ImplicitHeadMiddlewareTest.php]: I just realized this one is incorrect; we should likely be using POST or PATCH here; anything BUT HEAD or GET.

Comment (webimpress) [reply test/Middleware/ImplicitHeadMiddlewareTest.php]: Hm... It's fine, because this is case when we have implicit head `true`

Comment (weierophinney) [reply test/Middleware/ImplicitHeadMiddlewareTest.php]: Yeah, just came to that same conclusion. :+1:

**Pull request merged**